src/odoc/bin/main.ml

```diff
@@ -748,7 +778,8 @@ module Targets = struct
 
     let cmd = Term.(const list_targets $ Compile.dst $ Compile.input)
 
-    let info = Term.info "compile-targets" ~doc:"TODO: Fill in."
+    let info =
```

Comment:
Yes it does. 

Not sure what's the `dune` comment is about, but in general it's good for build systems if they can ask the tools which paths they are going to read and write, rather than have build tools guess, discover or hardcode the file systems effects.

(A good example of this is `support-files-targets`, if all goes well odig will have nothing to do to support the new math support)

Reply:
I don't see the point of this command as there's no reason for the compile command to output several files. Especially that it takes a `-o`.

Reply:
Because there's value in operational genericity. If all tools worked that way for all their steps, build incrementalization logics would be fully generic.

This benefits both build system (no tool specific logic) and the tools themselves (they can change the way they want to store intermediate results on the file system without having to bother about existing build logics).

Reply:
The output of `compile` must be given to `link` and the build system has to know that. I think this adds untested code paths and complexity rather than generality.

The build system must be changed every time Odoc changes so it has to stay simple to be able to adapt fast. Jon mentions Dune because it is built in a way that dynamic targets are hard to represent (a command that outputs an unknown number of files). It's trivial however to have a staticly known list of inputs and outputs.

Reply:
> The output of compile must be given to link and the build system has to know that.

That's not the point… It's about the incrementality logic (how do I determine if that step can be skipped and what effects do I need to perform or checked on the file system to make it as if it had been done) not the sequentiality logic (which sequence of operations do I need to invoke in order to get my final build result) which remains bound to the domain specific thing your are building.

Reply:
The "incrementality logic" should have a notion of statically known inputs/outputs, right ?

I still don't see why it's Odoc's job to say that `odoc compile -o foo bar` will write the file `foo` and read the file `bar`. Most compilers work that way and I've never seen a `-targets` command before.

Reply:
> The "incrementality logic" should have a notion of statically known inputs/outputs, right ?

No, `odoc html` is a good example. 

> I still don't see why it's Odoc's job to say that odoc compile -o foo bar will write the file foo and read the file bar. 

So who's job is it ? `odoc`'s documentation ? Why not have that machine readable ? 

If you want to make it easy for build systems to incrementalise builds made via stepwise invocation of cli tools, each compilation step should make it clear for a given invocation: which files it's going to read, which files it's going to write and which environment variables affect its outputs[^0]. Then you can simply see the incrementalization task as the memoization of pure functions (the cli invocations). 

> Most compilers work that way and I've never seen a -targets command before.

Most compilers do not make the job of build system writers easy.

[^0]: Writing a manifest of that after the operation was invoked is another option since determining actual outputs may be equivalent to run the action.

Reply:
We're talking about `compile-targets` here, not `html-targets`. 

> So who's job is it ?

It's the job of the "domain specific thing your are building" (that you opposed to "incrementality logic" in your other comment). 
I guess the person writing the "domain specific thing your are building" has read Odoc's documentation, so yes, it should be here too.

> each compilation step should make it clear for a given invocation: which files it's going to read, which files it's going to write and which environment variables affect its outputs1

That's the job of "domain specific thing your are building". At some point you have to write something like "for an input x and an output y, execute `odoc compile -o $y $x`. This information is already in your system. You also have to make sense of what `x` and `y` are, for example `y` is the input of other `odoc` commands that you define in the same system while `x` is from a complex analysis you've done before (perhaps using `odoc compile-deps` but not only that).

I don't see the point of pretending `odoc compile-targets -o $y ...` can return something else than `$y`. It makes other system a lot harder to write (eg. Dune).

Reply:
> We're talking about compile-targets here, not html-targets.

The discussion is running in circle. [Goto](https://github.com/ocaml/odoc/pull/903#discussion_r1026555387).

Reply:
Your comment doesn't explain why it's such a hassle to express that `odoc compile -o foo` will write `foo` but it's convenient to express that it will output the result of `odoc compile-deps -o foo`. 

You have to describe somewhere that `odoc compile-deps` exists. Please implement it like that in your own DSL:
```
let compile_targets output = [ output ]
```

The discussion is running in circle because you refuse to answer my point. We are both talking in the void, it's your fault.

Reply:
> We are both talking in the void, it's your fault.

Coming from you I will happily take all the blame.

Reply:
Please keep the discussion on the code not the people. We're not planning on removing the subcommand, so the discussion is purely about whether we document it in the legacy section or the support section. Since we have at least one user -- odig -- and there is no alternative other than assuming odoc's behaviour, let's put it in the support section.

Reply:
You are right, I moved the command to the support section.